### PR TITLE
Update example notebooks

### PR DESCRIPTION
**Pull request by jasongrout**
This is some light editing of the notebooks, correcting grammar, making the narrative flow better, checking examples, etc.

I'm not done yet!

- [ ] In the Custom Widget notebook, the very last cell, clicking on the spinner up/down arrows updates the spinner, but not the slider, until the spinner loses focus (in Chrome, for me).  When the spinner loses focus (i.e., the change event fires?), then the slider is updated...
- [ ] In the Widget Styling notebook, the container.pack example does not center the buttons like it says it will.

### CODE REVIEW

File: examples/Custom Widget - Hello World.ipynb [hunk 1]
```diff
@@ -771,21 +771,21 @@
  ],
  "metadata": {
   "kernelspec": {
-   "display_name": "Python 3",
+   "display_name": "Python 2",
```
Comment (jdfreder) [inline examples/Custom Widget - Hello World.ipynb]: These notebooks are supposed to Python 3.

Comment (jasongrout) [inline examples/Custom Widget - Hello World.ipynb]: Really?  Okay.  I guess that's an easy change.

Comment (jdfreder) [inline examples/Custom Widget - Hello World.ipynb]: If you'd like I can go back through them for you and do the conversion, just let me know.

**Comment by jasongrout** [comment]
Okay, I think I'm done with my pass, except for changing everything back to python 3.

**Comment by jasongrout** [comment]
@jdfreder, @SylvainCorlay - ready for review.

File: examples/Using Interact.ipynb [hunk 2]
```diff
@@ -32,7 +32,7 @@
    "metadata": {},
    "source": [
     "<div class=\"alert alert-success\">\n",
-    "As of IPython 3.0, the widgets in this notebook won't show up on http://nbviewer.ipython.org. To view the widgets and interact with them, you will need to download this notebook and run it with an IPython Notebook server.\n",
+    "As of IPython 3.0, the widgets in this notebook won't show up on http://nbviewer.ipython.org. To view the widgets and interact with them, you will need to download this notebook and run it with an IPython or Jupyter Notebook server.\n",
```
Comment (jdfreder) [inline examples/Using Interact.ipynb]: Should bump IPython 3.0 to 4.0

File: examples/Widget Basics.ipynb [hunk 3]
```diff
@@ -29,7 +29,7 @@
     }
    },
    "source": [
-    "Widgets are elements that exists in both the front-end and the back-end."
+    "Widgets are eventful python objects that have a representation in the browser, often as a control like a slider, textbox, etc."
```
Comment (jdfreder) [inline examples/Widget Basics.ipynb]: Nice, I like this description :+1:

**Comment by jdfreder** [comment]
Other than the tiny 3.0 -> 4.0 comment, this looks good to me.  @SylvainCorlay ?

**Comment by jasongrout** [comment]
Oh, and I still have the two bug reports in the issue description

File: examples/Using Interact.ipynb (hunk 2 above)
Comment (jasongrout) [inline examples/Using Interact.ipynb]: When I read it, I read it as a lower bound.  I see now that it's an upper bound.

**Pull request merged**